src/settings/handlers/RoomSettingsHandler.ts

```diff
@@ -92,12 +92,12 @@ export default class RoomSettingsHandler extends MatrixClientBackedSettingsHandl
         if (settingName === "urlPreviewsEnabled") {
             const content = this.getSettings(roomId, "org.matrix.room.preview_urls") || {};
             content['disable'] = !newValue;
-            return MatrixClientPeg.get().sendStateEvent(roomId, "org.matrix.room.preview_urls", content);
+            return MatrixClientPeg.get().sendStateEvent(roomId, "org.matrix.room.preview_urls", content).then();
```

Comment:
What are those `then` achieving?

Reply:
matching the signature, which is `Promise<void>` but sendStateEvent is `Promise<{ event_id: string }>`